stackslib/src/chainstate/nakamoto/staging_blocks.rs

```diff
@@ -206,26 +242,31 @@ impl<'a> NakamotoStagingBlocksConnRef<'a> {
         Ok(res.is_some())
     }
 
-    /// Get a staged Nakamoto tenure-start block
-    pub fn get_nakamoto_tenure_start_block(
+    /// Get the block ID, processed-status, orphan-status, and signing weight of the non-orphaned
+    /// block with the given consensus hash and sighash with the most amount of signatures.
+    /// There will be at most one such block.
+    ///
+    /// NOTE: for Nakamoto blocks, the sighash is the same as the block hash.
+    pub(crate) fn get_block_processed_and_signed_weight(
         &self,
         consensus_hash: &ConsensusHash,
-    ) -> Result<Option<NakamotoBlock>, ChainstateError> {
-        let qry = "SELECT data FROM nakamoto_staging_blocks WHERE is_tenure_start = 1 AND consensus_hash = ?1";
-        let args: &[&dyn ToSql] = &[consensus_hash];
-        let data: Option<Vec<u8>> = query_row(self, qry, args)?;
-        let Some(block_bytes) = data else {
-            return Ok(None);
-        };
-        let block = NakamotoBlock::consensus_deserialize(&mut block_bytes.as_slice())?;
-        if &block.header.consensus_hash != consensus_hash {
-            error!(
-                "Staging DB corruption: expected {}, got {}",
-                consensus_hash, block.header.consensus_hash
-            );
-            return Err(DBError::Corruption.into());
+        sighash: &Sha512Trunc256Sum,
```

Comment:
I don't think we should use the term sighash anywhere to refer to the signer signature hash. `signer_sighash`, maybe. But there's at least two "signature hashes" that could be referred to in a stacks block.

Reply:
Sure, `signer_sighash` it is